src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -120,7 +120,10 @@
           col_vals <- levels(val)
         }
       }
-      else if (is.numeric(val))
+      # for histograms, we support only the base R numeric class and its derivatives;
+      # is.numeric can return true for values that can only be manipulated using
+      # packages that are currently loaded (e.g. bit64's integer64)
+      else if (is.numeric(val) && inherits(x[[idx]], "numeric"))
```

Comment:
Do we want to permit integer vectors here?

```R
> inherits(1L, "numeric")
[1] FALSE
```

If we want to only allow 'true' primitive integer / numeric vectors, you might instead try checking `is.object()`, e.g.

```R
> is.object(1)
[1] FALSE
> is.object(1L)
[1] FALSE
> is.object(bit64::integer64())
[1] TRUE
```

`is.object()` is basically a way of seeing if the object has an explicit S3 or S4 class registered.

Reply:
Yes, I think `is.object` makes more sense here! I've just pushed the change.